### ISSUE TITLE
`zppy` cannot discover the hostname from a compute node on chrysalis

### ISSUE THREAD

**Issue by mahf708** [E3SM-Project/zppy#406]
```
[ac.ngmahfouz@chr-0493 pipd_cdf6c6e]$ source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh
(e3sm_unified_1.8.0_chrysalis) [ac.ngmahfouz@chr-0493 pipd_cdf6c6e]$ source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.shD^C
(e3sm_unified_1.8.0_chrysalis) [ac.ngmahfouz@chr-0493 pipd_cdf6c6e]$ zppy -c F2010.PD.NGD_v3atm.9ac9529.chrysalis.cfg 
Configuration file validation passed.
Traceback (most recent call last):
  File "/lcrc/soft/climate/e3sm-unified/base/envs/e3sm_unified_1.8.0_chrysalis/bin/zppy", line 10, in <module>
    sys.exit(main())
  File "/lcrc/soft/climate/e3sm-unified/base/envs/e3sm_unified_1.8.0_chrysalis/lib/python3.10/site-packages/zppy/__main__.py", line 84, in main
    machine_info = MachineInfo(machine=machine)
  File "/lcrc/soft/climate/e3sm-unified/base/envs/e3sm_unified_1.8.0_chrysalis/lib/python3.10/site-packages/mache/machine_info.py", line 75, in __init__
    raise ValueError('Unable to discover machine from host name')
ValueError: Unable to discover machine from host name
```

**Comment by xylar** [comment]
@mahf708, as discussed in https://github.com/E3SM-Project/mache/issues/58.  This really is a zppy problem.  zppy needs to detect the machine when you're still on a login node and set the machine manually on compute nodes.

**Comment by xylar** [comment]
I don't think there's any way for mache to handle this so it's a zppy problem if this should work when you launch jobs from a compute node.

**Comment by xylar** [comment]
@mahf708, if you use the development version of zppy, you should be able to set the machine as a config option and then launch the job on a compute node.  That would be my suggestion.  I don't think @forsyth2 was interested in an automated approach but one other possibility specifically in E3SM-Unified would be to use the `$E3SMU_MACHINE` variable.

**Comment by xylar** [comment]
@forsyth2, could you point @mahf708 to an example config file that has the `machine` config option?

**Comment by forsyth2** [comment]
> I don't think @forsyth2 was interested in an automated approach

I think the issue was that on some machines it's difficult to tell what machine you're on based on the compute-node name.

> @forsyth2, could you point @mahf708 to an example config file that has the `machine` config option?

I don't seem to have any at the moment, but you can just set `machine =` to `chrysalis`. (https://github.com/E3SM-Project/zppy/blob/main/zppy/__main__.py#L111 shows options for different machines).

> a good habit to quickly report quirks

Yes, I agree it's good to document issues and/or feature requests. That said, it's helpful to know if an issue requires immediate attention or not. I just added the "Nice to have" label for this issue. That label tells me I can prioritize other tasks and work on an issue time-permitting (or if enough people request it that it's no longer just "nice to have").

**Comment by xylar** [comment]
@mahf708, I really think the desired solution to this is that you add:
```
[default]
machine = chrysalis
```
(or, obviously, the equivalent for other machines) to your config file if you want to launch a job from a compute node.  We can fix this so it automatically detects the machine in E3SM-Unified for the next release, but that won't help you with the current release or with zppy development environments in the meantime.

**Comment by xylar** [comment]
I have fixed machine discovery with E3SM-Unified in #511
